acceptance-tests/src/main/java/io/blueocean/ath/pages/blue/ActivityPage.java

```diff
@@ -87,6 +90,12 @@ public BranchPage clickBranchTab() {
         return branchPageFactory.withPipeline(pipeline).checkUrl();
     }
 
+    public PullRequestsPage clickPullRequestsTab() {
+        wait.until(ExpectedConditions.elementToBeClickable(By.cssSelector("a.pr"))).click();
+        logger.info("Clicked on PR tab");
+        return pullRequestsPageFactory.withPipeline(pipeline).checkUrl();
```

Comment:
Any possibilities of `withPipeline` returning null? If that is the case it will be easier to debug if you split this line in something like:

```
PullRequestsPage page = pullRequestsPageFactory.withPipeline(pipeline)
assertNotNull("Whatever error message", page)
return page.checkUrl()
```

It does not make the test more robust but it makes it easier to debug in case of failure

Reply:
Great suggestion - anything's possible. :)  I actually ran into a NPE very much like what you're talking about when working on this test. It wasn't from this specifically, but seeing a useful error message rather than an NPE would have probably saved me some heartache. I've made the change, and have mimicked it in the method just above it, `clickBranchTab()`.